### ISSUE TITLE
ANR com.onesignal.SyncService

### ISSUE THREAD

**Issue by devlab-dariolipari** [OneSignal/OneSignal-Android-SDK#589]
I notice this ANR on Google Play Console:
```java
"main" prio=5 tid=1 MONITOR
  | group="main" sCount=1 dsCount=0 obj=0x41652cc0 self=0x4158dd80
  | sysTid=3640 nice=-11 sched=0/0 cgrp=apps handle=1073992020
  | state=S schedstat=( 0 0 0 ) utm=6 stm=10 core=0
  at com.onesignal.OneSignal.onAppFocus (OneSignal.java:1113)
  at com.onesignal.ActivityLifecycleHandler.handleFocus (ActivityLifecycleHandler.java:120)
  at com.onesignal.ActivityLifecycleHandler.onActivityResumed (ActivityLifecycleHandler.java:74)
  at com.onesignal.ActivityLifecycleListener.onActivityResumed (ActivityLifecycleListener.java:48)
  at android.app.Application.dispatchActivityResumed (Application.java:228)
  at android.app.Activity.onResume (Activity.java:1149)
  at android.app.Instrumentation.callActivityOnResume (Instrumentation.java:1199)
  at android.app.Activity.performResume (Activity.java:5521)
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:3046)
  at android.app.ActivityThread.handleResumeActivity (ActivityThread.java:3085)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2498)
  at android.app.ActivityThread.access$800 (ActivityThread.java:166)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1283)
  at android.os.Handler.dispatchMessage (Handler.java:102)
  at android.os.Looper.loop (Looper.java:136)
  at android.app.ActivityThread.main (ActivityThread.java:5584)
  at java.lang.reflect.Method.invokeNative (Native Method)
  at java.lang.reflect.Method.invoke (Method.java:515)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1268)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1084)
  at dalvik.system.NativeStart.main (Native Method)
```

**Comment by jkasten2** [comment]
@devlab-dariolipari Thanks for the report. Can you include the full ANR trace for all threads? We will need this to see what state the other threads are in your app to determine exactly how the lock occurred.

**Comment by devlab-dariolipari** [comment]
@jkasten2 thanks for the reply.

Here the full ANR trace.

---------------------------------------------------------------------------------------------------

"main" prio=5 tid=1 MONITOR
  | group="main" sCount=1 dsCount=0 obj=0x41652cc0 self=0x4158dd80
  | sysTid=3640 nice=-11 sched=0/0 cgrp=apps handle=1073992020
  | state=S schedstat=( 0 0 0 ) utm=6 stm=10 core=0
  at com.onesignal.OneSignal.onAppFocus (OneSignal.java:1113)
  at com.onesignal.ActivityLifecycleHandler.handleFocus (ActivityLifecycleHandler.java:120)
  at com.onesignal.ActivityLifecycleHandler.onActivityResumed (ActivityLifecycleHandler.java:74)
  at com.onesignal.ActivityLifecycleListener.onActivityResumed (ActivityLifecycleListener.java:48)
  at android.app.Application.dispatchActivityResumed (Application.java:228)
  at android.app.Activity.onResume (Activity.java:1149)
  at android.app.Instrumentation.callActivityOnResume (Instrumentation.java:1199)
  at android.app.Activity.performResume (Activity.java:5521)
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:3046)
  at android.app.ActivityThread.handleResumeActivity (ActivityThread.java:3085)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2498)
  at android.app.ActivityThread.access$800 (ActivityThread.java:166)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1283)
  at android.os.Handler.dispatchMessage (Handler.java:102)
  at android.os.Looper.loop (Looper.java:136)
  at android.app.ActivityThread.main (ActivityThread.java:5584)
  at java.lang.reflect.Method.invokeNative (Native Method)
  at java.lang.reflect.Method.invoke (Method.java:515)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1268)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1084)
  at dalvik.system.NativeStart.main (Native Method)
"GC" tid=2 WaitingForVm 
"GC" daemon prio=5 tid=2 VMWAIT
  | group="system" sCount=1 dsCount=0 obj=0x41dfdd90 self=0x50cfa010
  | sysTid=3642 nice=0 sched=0/0 cgrp=apps handle=1096344800
  | state=S schedstat=( 0 0 0 ) utm=0 stm=0 core=0
  #00  pc 00000000000219ac  /system/lib/libc.so (__futex_syscall3+8)
  #01  pc 000000000000efdc  /system/lib/libc.so (__pthread_cond_timedwait_relative+48)
  #02  pc 000000000000f03c  /system/lib/libc.so (__pthread_cond_timedwait+64)
  #03  pc 0000000000072a5b  /system/lib/libdvm.so
  #04  pc 0000000000054a75  /system/lib/libdvm.so
  #05  pc 000000000000d280  /system/lib/libc.so (__thread_entry+72)
  #06  pc 000000000000d41c  /system/lib/libc.so (pthread_create+240)
  at dalvik.system.NativeStart.run (Native Method)
"Compiler" tid=4 WaitingForVm 
"Compiler" daemon prio=5 tid=4 VMWAIT
  | group="system" sCount=1 dsCount=0 obj=0x41dfdf68 self=0x5101e968
  | sysTid=3644 nice=0 sched=0/0 cgrp=apps handle=1336309352
  | state=S schedstat=( 0 0 0 ) utm=8 stm=2 core=2
  #00  pc 00000000000219ac  /system/lib/libc.so (__futex_syscall3+8)
  #01  pc 000000000000efdc  /system/lib/libc.so (__pthread_cond_timedwait_relative+48)
  #02  pc 000000000000f03c  /system/lib/libc.so (__pthread_cond_timedwait+64)
  #03  pc 00000000000743cf  /system/lib/libdvm.so
  #04  pc 0000000000054a75  /system/lib/libdvm.so
  #05  pc 000000000000d280  /system/lib/libc.so (__thread_entry+72)
  #06  pc 000000000000d41c  /system/lib/libc.so (pthread_create+240)
  at dalvik.system.NativeStart.run (Native Method)
"ReferenceQueueDaemon" tid=5 Waiting 
"ReferenceQueueDaemon" daemon prio=5 tid=5 WAIT
  | group="system" sCount=1 dsCount=0 obj=0x41dfe058 self=0x51254470
  | sysTid=3645 nice=0 sched=0/0 cgrp=apps handle=1359081416
  | state=S schedstat=( 0 0 0 ) utm=0 stm=0 core=0
  at java.lang.Object.wait (Native Method)
- waiting on <0x41647640>
  at java.lang.Object.wait (Object.java:364)
  at java.lang.Daemons$ReferenceQueueDaemon.run (Daemons.java:130)
  at java.lang.Thread.run (Thread.java:841)
"FinalizerDaemon" tid=6 Waiting 
"FinalizerDaemon" daemon prio=5 tid=6 WAIT
  | group="system" sCount=1 dsCount=0 obj=0x41dfe1c0 self=0x51254ad0
  | sysTid=3646 nice=0 sched=0/0 cgrp=apps handle=1361399592
  | state=S schedstat=( 0 0 0 ) utm=0 stm=0 core=0
  at java.lang.Object.wait (Native Method)
- waiting on <0x41647718> (a java.lang.ref.ReferenceQueue)
  at java.lang.Object.wait (Object.java:401)
  at java.lang.ref.ReferenceQueue.remove (ReferenceQueue.java:102)
  at java.lang.ref.ReferenceQueue.remove (ReferenceQueue.java:73)
  at java.lang.Daemons$FinalizerDaemon.run (Daemons.java:172)
  at java.lang.Thread.run (Thread.java:841)
"FinalizerWatchdogDaemon" tid=7 Waiting 
"FinalizerWatchdogDaemon" daemon prio=5 tid=7 WAIT
  | group="system" sCount=1 dsCount=0 obj=0x41dfe318 self=0x51255378
  | sysTid=3647 nice=0 sched=0/0 cgrp=apps handle=1361401808
  | state=S schedstat=( 0 0 0 ) utm=0 stm=0 core=0
  at java.lang.Object.wait (Native Method)
- waiting on <0x4165c430> (a java.lang.Daemons$FinalizerWatchdogDaemon)
  at java.lang.Object.wait (Object.java:364)
  at java.lang.Daemons$FinalizerWatchdogDaemon.run (Daemons.java:229)
  at java.lang.Thread.run (Thread.java:841)
"Binder_1" tid=8 Native 
"Binder_1" prio=5 tid=8 NATIVE
  | group="main" sCount=1 dsCount=0 obj=0x41e042c0 self=0x51078320
  | sysTid=3648 nice=0 sched=0/0 cgrp=apps handle=1361403176
  | state=S schedstat=( 0 0 0 ) utm=1 stm=0 core=2
  #00  pc 0000000000020630  /system/lib/libc.so (__ioctl+8)
  #01  pc 000000000002d05f  /system/lib/libc.so (ioctl+14)
  #02  pc 000000000001e59d  /system/lib/libbinder.so (android::IPCThreadState::talkWithDriver(bool)+140)
  #03  pc 000000000001eceb  /system/lib/libbinder.so (android::IPCThreadState::getAndExecuteCommand()+6)
  #04  pc 000000000001ed81  /system/lib/libbinder.so (android::IPCThreadState::joinThreadPool(bool)+48)
  #05  pc 0000000000023431  /system/lib/libbinder.so
  #06  pc 000000000000ea61  /system/lib/libutils.so (android::Thread::_threadLoop(void*)+216)
  #07  pc 000000000004ecf9  /system/lib/libandroid_runtime.so (android::AndroidRuntime::javaThreadShell(void*)+68)
  #08  pc 000000000000e593  /system/lib/libutils.so
  #09  pc 000000000000d280  /system/lib/libc.so (__thread_entry+72)
  #10  pc 000000000000d41c  /system/lib/libc.so (pthread_create+240)
  at dalvik.system.NativeStart.run (Native Method)
"Binder_2" tid=9 Native 
"Binder_2" prio=5 tid=9 NATIVE
  | group="main" sCount=1 dsCount=0 obj=0x41e081a8 self=0x5118cc58
  | sysTid=3651 nice=0 sched=0/0 cgrp=apps handle=1360576840
  | state=S schedstat=( 0 0 0 ) utm=0 stm=0 core=2
  #00  pc 0000000000020630  /system/lib/libc.so (__ioctl+8)
  #01  pc 000000000002d05f  /system/lib/libc.so (ioctl+14)
  #02  pc 000000000001e59d  /system/lib/libbinder.so (android::IPCThreadState::talkWithDriver(bool)+140)
  #03  pc 000000000001eceb  /system/lib/libbinder.so (android::IPCThreadState::getAndExecuteCommand()+6)
  #04  pc 000000000001ed81  /system/lib/libbinder.so (android::IPCThreadState::joinThreadPool(bool)+48)
  #05  pc 0000000000023431  /system/lib/libbinder.so
  #06  pc 000000000000ea61  /system/lib/libutils.so (android::Thread::_threadLoop(void*)+216)
  #07  pc 000000000004ecf9  /system/lib/libandroid_runtime.so (android::AndroidRuntime::javaThreadShell(void*)+68)
  #08  pc 000000000000e593  /system/lib/libutils.so
  #09  pc 000000000000d280  /system/lib/libc.so (__thread_entry+72)
  #10  pc 000000000000d41c  /system/lib/libc.so (pthread_create+240)
  at dalvik.system.NativeStart.run (Native Method)
"OSH_WritePrefs" tid=10 Native 
"OSH_WritePrefs" prio=5 tid=10 NATIVE
  | group="main" sCount=1 dsCount=0 obj=0x41e339c0 self=0x511b5010
  | sysTid=3653 nice=0 sched=0/0 cgrp=apps handle=1360745576
  | state=S schedstat=( 0 0 0 ) utm=0 stm=0 core=0
  #00  pc 00000000000217e4  /system/lib/libc.so (epoll_wait+12)
  #01  pc 00000000000107a7  /system/lib/libutils.so (android::Looper::pollInner(int)+94)
  #02  pc 00000000000109d5  /system/lib/libutils.so (android::Looper::pollOnce(int, int*, int*, void**)+92)
  #03  pc 000000000006d2cd  /system/lib/libandroid_runtime.so (android::NativeMessageQueue::pollOnce(_JNIEnv*, int)+22)
  #04  pc 000000000001dd8c  /system/lib/libdvm.so (dvmPlatformInvoke+112)
  #05  pc 000000000004e253  /system/lib/libdvm.so (dvmCallJNIMethod(unsigned int const*, JValue*, Method const*, Thread*)+398)
  #06  pc 00000000000271a0  /system/lib/libdvm.so
  #07  pc 000000000002e150  /system/lib/libdvm.so (dvmMterpStd(Thread*)+76)
  #08  pc 000000000002b7fc  /system/lib/libdvm.so (dvmInterpret(Thread*, Method const*, JValue*)+184)
  #09  pc 0000000000060843  /system/lib/libdvm.so (dvmCallMethodV(Thread*, Method const*, Object*, bool, JValue*, std::__va_list)+338)
  #10  pc 0000000000060867  /system/lib/libdvm.so (dvmCallMethod(Thread*, Method const*, Object*, JValue*, ...)+20)
  #11  pc 0000000000055547  /system/lib/libdvm.so
  #12  pc 000000000000d280  /system/lib/libc.so (__thread_entry+72)
  #13  pc 000000000000d41c  /system/lib/libc.so (pthread_create+240)
  at android.os.MessageQueue.nativePollOnce (Native Method)
  at android.os.MessageQueue.next (MessageQueue.java:138)
  at android.os.Looper.loop (Looper.java:123)
  at android.os.HandlerThread.run (HandlerThread.java:61)
"FocusHandlerThread" tid=11 Native 
"FocusHandlerThread" prio=5 tid=11 NATIVE
  | group="main" sCount=1 dsCount=0 obj=0x41e38190 self=0x511b58c0
  | sysTid=3654 nice=0 sched=0/0 cgrp=apps handle=1360747800
  | state=S schedstat=( 0 0 0 ) utm=0 stm=0 core=0
  #00  pc 00000000000217e4  /system/lib/libc.so (epoll_wait+12)
  #01  pc 00000000000107a7  /system/lib/libutils.so (android::Looper::pollInner(int)+94)
  #02  pc 00000000000109d5  /system/lib/libutils.so (android::Looper::pollOnce(int, int*, int*, void**)+92)
  #03  pc 000000000006d2cd  /system/lib/libandroid_runtime.so (android::NativeMessageQueue::pollOnce(_JNIEnv*, int)+22)
  #04  pc 000000000001dd8c  /system/lib/libdvm.so (dvmPlatformInvoke+112)
  #05  pc 000000000004e253  /system/lib/libdvm.so (dvmCallJNIMethod(unsigned int const*, JValue*, Method const*, Thread*)+398)
  #06  pc 00000000000271a0  /system/lib/libdvm.so
  #07  pc 000000000002e150  /system/lib/libdvm.so (dvmMterpStd(Thread*)+76)
  #08  pc 000000000002b7fc  /system/lib/libdvm.so (dvmInterpret(Thread*, Method const*, JValue*)+184)
  #09  pc 0000000000060843  /system/lib/libdvm.so (dvmCallMethodV(Thread*, Method const*, Object*, bool, JValue*, std::__va_list)+338)
  #10  pc 0000000000060867  /system/lib/libdvm.so (dvmCallMethod(Thread*, Method const*, Object*, JValue*, ...)+20)
  #11  pc 0000000000055547  /system/lib/libdvm.so
  #12  pc 000000000000d280  /system/lib/libc.so (__thread_entry+72)
  #13  pc 000000000000d41c  /system/lib/libc.so (pthread_create+240)
  at android.os.MessageQueue.nativePollOnce (Native Method)
  at android.os.MessageQueue.next (MessageQueue.java:138)
  at android.os.Looper.loop (Looper.java:123)
  at android.os.HandlerThread.run (HandlerThread.java:61)
"OSH_NetworkHandlerThread" tid=12 Native 
"OSH_NetworkHandlerThread" prio=5 tid=12 NATIVE
  | group="main" sCount=1 dsCount=0 obj=0x41e5e6c0 self=0x5123e210
  | sysTid=3656 nice=0 sched=0/0 cgrp=apps handle=1361307240
  | state=S schedstat=( 0 0 0 ) utm=0 stm=0 core=0
  #00  pc 00000000000217e4  /system/lib/libc.so (epoll_wait+12)
  #01  pc 00000000000107a7  /system/lib/libutils.so (android::Looper::pollInner(int)+94)
  #02  pc 00000000000109d5  /system/lib/libutils.so (android::Looper::pollOnce(int, int*, int*, void**)+92)
  #03  pc 000000000006d2cd  /system/lib/libandroid_runtime.so (android::NativeMessageQueue::pollOnce(_JNIEnv*, int)+22)
  #04  pc 000000000001dd8c  /system/lib/libdvm.so (dvmPlatformInvoke+112)
  #05  pc 000000000004e253  /system/lib/libdvm.so (dvmCallJNIMethod(unsigned int const*, JValue*, Method const*, Thread*)+398)
  #06  pc 00000000000271a0  /system/lib/libdvm.so
  #07  pc 000000000002e150  /system/lib/libdvm.so (dvmMterpStd(Thread*)+76)
  #08  pc 000000000002b7fc  /system/lib/libdvm.so (dvmInterpret(Thread*, Method const*, JValue*)+184)
  #09  pc 0000000000060843  /system/lib/libdvm.so (dvmCallMethodV(Thread*, Method const*, Object*, bool, JValue*, std::__va_list)+338)
  #10  pc 0000000000060867  /system/lib/libdvm.so (dvmCallMethod(Thread*, Method const*, Object*, JValue*, ...)+20)
  #11  pc 0000000000055547  /system/lib/libdvm.so
  #12  pc 000000000000d280  /system/lib/libc.so (__thread_entry+72)
  #13  pc 000000000000d41c  /system/lib/libc.so (pthread_create+240)
  at android.os.MessageQueue.nativePollOnce (Native Method)
  at android.os.MessageQueue.next (MessageQueue.java:138)
  at android.os.Looper.loop (Looper.java:123)
  at android.os.HandlerThread.run (HandlerThread.java:61)
"GAC_Executor[1]" tid=13 Waiting 
"GAC_Executor[1]" prio=5 tid=13 WAIT
  | group="main" sCount=1 dsCount=0 obj=0x41e90140 self=0x5123f5e0
  | sysTid=3661 nice=0 sched=0/0 cgrp=apps handle=1361312312
  | state=S schedstat=( 0 0 0 ) utm=0 stm=0 core=0
  at java.lang.Object.wait (Native Method)
- waiting on <0x41e902e0> (a java.lang.VMThread) held by tid=13 (GAC_Executor[1])
  at java.lang.Thread.parkFor (Thread.java:1205)
  at sun.misc.Unsafe.park (Unsafe.java:325)
  at java.util.concurrent.locks.LockSupport.park (LockSupport.java:157)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await (AbstractQueuedSynchronizer.java:2017)
  at java.util.concurrent.LinkedBlockingQueue.take (LinkedBlockingQueue.java:410)
  at java.util.concurrent.ThreadPoolExecutor.getTask (ThreadPoolExecutor.java:1035)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1097)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:587)
  at java.lang.Thread.run (Thread.java:841)
"OS_SYNCSRV_BG_SYNC" tid=14 TimedWaiting 
"OS_SYNCSRV_BG_SYNC" prio=5 tid=14 TIMED_WAIT
  | group="main" sCount=1 dsCount=0 obj=0x42065550 self=0x51246f08
  | sysTid=4241 nice=0 sched=0/0 cgrp=apps handle=1361324744
  | state=S schedstat=( 0 0 0 ) utm=0 stm=0 core=0
  at java.lang.Object.wait (Native Method)
- waiting on <0x4206baa0> (a java.lang.VMThread) held by tid=19 (OS_HTTPConnection)
  at java.lang.Thread.join (Thread.java:809)
  at java.lang.Thread.join (Thread.java:765)
  at com.onesignal.OneSignalRestClient.makeRequest (OneSignalRestClient.java:106)
  at com.onesignal.OneSignalRestClient.putSync (OneSignalRestClient.java:82)
  at com.onesignal.UserStateSynchronizer.doPutSync (UserStateSynchronizer.java:255)
  at com.onesignal.UserStateSynchronizer.internalSyncUserState (UserStateSynchronizer.java:189)
  at com.onesignal.UserStateSynchronizer.syncUserState (UserStateSynchronizer.java:160)
  at com.onesignal.OneSignalStateSynchronizer.syncUserState (OneSignalStateSynchronizer.java:70)
  at com.onesignal.OneSignalSyncServiceUtils$SyncRunnable$1.complete (OneSignalSyncServiceUtils.java:228)
  at com.onesignal.LocationGMS.fireComplete (LocationGMS.java:254)
  at com.onesignal.LocationGMS.fireCompleteForLocation (LocationGMS.java:288)
  at com.onesignal.LocationGMS.startGetLocation (LocationGMS.java:202)
  at com.onesignal.LocationGMS.getLocation (LocationGMS.java:147)
  at com.onesignal.OneSignalSyncServiceUtils$SyncRunnable.run (OneSignalSyncServiceUtils.java:233)
  at java.lang.Thread.run (Thread.java:841)
"OSH_LocationHandlerThread" tid=15 Native 
"OSH_LocationHandlerThread" prio=5 tid=15 NATIVE
  | group="main" sCount=1 dsCount=0 obj=0x41e68008 self=0x512435e8
  | sysTid=3659 nice=0 sched=0/0 cgrp=apps handle=1361328704
  | state=S schedstat=( 0 0 0 ) utm=1 stm=1 core=0
  #00  pc 00000000000217e4  /system/lib/libc.so (epoll_wait+12)
  #01  pc 00000000000107a7  /system/lib/libutils.so (android::Looper::pollInner(int)+94)
  #02  pc 00000000000109d5  /system/lib/libutils.so (android::Looper::pollOnce(int, int*, int*, void**)+92)
  #03  pc 000000000006d2cd  /system/lib/libandroid_runtime.so (android::NativeMessageQueue::pollOnce(_JNIEnv*, int)+22)
  #04  pc 000000000001dd8c  /system/lib/libdvm.so (dvmPlatformInvoke+112)
  #05  pc 000000000004e253  /system/lib/libdvm.so (dvmCallJNIMethod(unsigned int const*, JValue*, Method const*, Thread*)+398)
  #06  pc 00000000000271a0  /system/lib/libdvm.so
  #07  pc 000000000002e150  /system/lib/libdvm.so (dvmMterpStd(Thread*)+76)
  #08  pc 000000000002b7fc  /system/lib/libdvm.so (dvmInterpret(Thread*, Method const*, JValue*)+184)
  #09  pc 0000000000060843  /system/lib/libdvm.so (dvmCallMethodV(Thread*, Method const*, Object*, bool, JValue*, std::__va_list)+338)
  #10  pc 0000000000060867  /system/lib/libdvm.so (dvmCallMethod(Thread*, Method const*, Object*, JValue*, ...)+20)
  #11  pc 0000000000055547  /system/lib/libdvm.so
  #12  pc 000000000000d280  /system/lib/libc.so (__thread_entry+72)
  #13  pc 000000000000d41c  /system/lib/libc.so (pthread_create+240)
  at android.os.MessageQueue.nativePollOnce (Native Method)
  at android.os.MessageQueue.next (MessageQueue.java:138)
  at android.os.Looper.loop (Looper.java:123)
  at android.os.HandlerThread.run (HandlerThread.java:61)
"GAC_Executor[0]" tid=16 Waiting 
"GAC_Executor[0]" prio=5 tid=16 WAIT
  | group="main" sCount=1 dsCount=0 obj=0x41e85428 self=0x512445c0
  | sysTid=3660 nice=0 sched=0/0 cgrp=apps handle=1361330984
  | state=S schedstat=( 0 0 0 ) utm=0 stm=0 core=0
  at java.lang.Object.wait (Native Method)
- waiting on <0x41e85660> (a java.lang.VMThread) held by tid=16 (GAC_Executor[0])
  at java.lang.Thread.parkFor (Thread.java:1205)
  at sun.misc.Unsafe.park (Unsafe.java:325)
  at java.util.concurrent.locks.LockSupport.park (LockSupport.java:157)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await (AbstractQueuedSynchronizer.java:2017)
  at java.util.concurrent.LinkedBlockingQueue.take (LinkedBlockingQueue.java:410)
  at java.util.concurrent.ThreadPoolExecutor.getTask (ThreadPoolExecutor.java:1035)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1097)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:587)
  at java.lang.Thread.run (Thread.java:841)
"OS_GMS_LOCATION_FALLBACK" tid=17 Blocked 
"OS_GMS_LOCATION_FALLBACK" prio=5 tid=17 MONITOR
  | group="main" sCount=1 dsCount=0 obj=0x42065970 self=0x572a96e8
  | sysTid=4242 nice=0 sched=0/0 cgrp=apps handle=1462410048
  | state=S schedstat=( 0 0 0 ) utm=0 stm=5 core=0
  at com.onesignal.LocationGMS$2.run (LocationGMS.java:220)
  at java.lang.Thread.run (Thread.java:841)
"pool-3-thread-1" tid=18 Waiting 
"pool-3-thread-1" prio=5 tid=18 WAIT
  | group="main" sCount=1 dsCount=0 obj=0x41eee320 self=0x57171de0
  | sysTid=3663 nice=0 sched=0/0 cgrp=apps handle=1461133880
  | state=S schedstat=( 0 0 0 ) utm=0 stm=2 core=0
  at java.lang.Object.wait (Native Method)
- waiting on <0x41eee468> (a java.lang.VMThread) held by tid=18 (pool-3-thread-1)
  at java.lang.Thread.parkFor (Thread.java:1205)
  at sun.misc.Unsafe.park (Unsafe.java:325)
  at java.util.concurrent.locks.LockSupport.park (LockSupport.java:157)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await (AbstractQueuedSynchronizer.java:2017)
  at java.util.concurrent.LinkedBlockingQueue.take (LinkedBlockingQueue.java:410)
  at java.util.concurrent.ThreadPoolExecutor.getTask (ThreadPoolExecutor.java:1035)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1097)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:587)
  at java.lang.Thread.run (Thread.java:841)
"OS_HTTPConnection" tid=19 Native 
"OS_HTTPConnection" prio=5 tid=19 NATIVE
  | group="main" sCount=1 dsCount=0 obj=0x4206b9e8 self=0x572dc8d8
  | sysTid=4243 nice=0 sched=0/0 cgrp=apps handle=1462619440
  | state=S schedstat=( 0 0 0 ) utm=0 stm=0 core=0
  #00  pc 0000000000020780  /system/lib/libc.so (select+20)
  #01  pc 000000000000a14d  /system/lib/libjavacrypto.so
  #02  pc 000000000000b1b9  /system/lib/libjavacrypto.so
  #03  pc 000000000001dd8c  /system/lib/libdvm.so (dvmPlatformInvoke+112)
  #04  pc 000000000004e253  /system/lib/libdvm.so (dvmCallJNIMethod(unsigned int const*, JValue*, Method const*, Thread*)+398)
  #05  pc 00000000000271a0  /system/lib/libdvm.so
  #06  pc 000000000002e150  /system/lib/libdvm.so (dvmMterpStd(Thread*)+76)
  #07  pc 000000000002b7fc  /system/lib/libdvm.so (dvmInterpret(Thread*, Method const*, JValue*)+184)
  #08  pc 0000000000060843  /system/lib/libdvm.so (dvmCallMethodV(Thread*, Method const*, Object*, bool, JValue*, std::__va_list)+338)
  #09  pc 0000000000060867  /system/lib/libdvm.so (dvmCallMethod(Thread*, Method const*, Object*, JValue*, ...)+20)
  #10  pc 0000000000055547  /system/lib/libdvm.so
  #11  pc 000000000000d280  /system/lib/libc.so (__thread_entry+72)
  #12  pc 000000000000d41c  /system/lib/libc.so (pthread_create+240)
  at com.android.org.conscrypt.NativeCrypto.SSL_read (Native Method)
  at com.android.org.conscrypt.OpenSSLSocketImpl$SSLInputStream.read (OpenSSLSocketImpl.java:689)
  at java.io.BufferedInputStream.fillbuf (BufferedInputStream.java:168)
  at java.io.BufferedInputStream.read (BufferedInputStream.java:227)
  at com.android.okhttp.internal.Util.readAsciiLine (Util.java:316)
  at com.android.okhttp.internal.http.RawHeaders.fromBytes (RawHeaders.java:308)
  at com.android.okhttp.internal.http.HttpTransport.readResponseHeaders (HttpTransport.java:135)
  at com.android.okhttp.internal.http.HttpEngine.readResponse (HttpEngine.java:644)
  at com.android.okhttp.internal.http.HttpURLConnectionImpl.execute (HttpURLConnectionImpl.java:347)
  at com.android.okhttp.internal.http.HttpURLConnectionImpl.getResponse (HttpURLConnectionImpl.java:296)
  at com.android.okhttp.internal.http.HttpURLConnectionImpl.getResponseCode (HttpURLConnectionImpl.java:503)
  at com.android.okhttp.internal.http.HttpsURLConnectionImpl.getResponseCode (HttpsURLConnectionImpl.java:136)
  at com.onesignal.OneSignalRestClient.startHTTPConnection (OneSignalRestClient.java:150)
  at com.onesignal.OneSignalRestClient.access$100 (OneSignalRestClient.java:38)
  at com.onesignal.OneSignalRestClient$4.run (OneSignalRestClient.java:98)
  at java.lang.Thread.run (Thread.java:841)
"AsyncTask #1" tid=21 Waiting 
"AsyncTask #1" prio=5 tid=21 WAIT
  | group="main" sCount=1 dsCount=0 obj=0x41fdd308 self=0x57285008
  | sysTid=4111 nice=10 sched=0/0 cgrp=apps/bg_non_interactive handle=1462233920
  | state=S schedstat=( 0 0 0 ) utm=2 stm=3 core=1
  at java.lang.Object.wait (Native Method)
- waiting on <0x41fdd4a0> (a java.lang.VMThread) held by tid=21 (AsyncTask #1)
  at java.lang.Thread.parkFor (Thread.java:1205)
  at sun.misc.Unsafe.park (Unsafe.java:325)
  at java.util.concurrent.locks.LockSupport.park (LockSupport.java:157)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await (AbstractQueuedSynchronizer.java:2017)
  at java.util.concurrent.LinkedBlockingQueue.take (LinkedBlockingQueue.java:410)
  at java.util.concurrent.ThreadPoolExecutor.getTask (ThreadPoolExecutor.java:1035)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1097)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:587)
  at java.lang.Thread.run (Thread.java:841)
"Signal Catcher" tid=3 Runnable 
"Signal Catcher" daemon prio=5 tid=3 RUNNABLE
  | group="system" sCount=0 dsCount=0 obj=0x41dfde70 self=0x4fa67008
  | sysTid=3643 nice=0 sched=0/0 cgrp=apps handle=1361395848
  | state=R schedstat=( 0 0 0 ) utm=2 stm=2 core=3
  at dalvik.system.NativeStart.run (Native Method)

**Comment by Nightsd01** [comment]
Resolved in 3.10.1! Thanks for reporting this issue
